mods/default/functions.lua

```diff
@@ -366,8 +364,7 @@ minetest.register_abm({
 		end
 
 		-- Look for likely neighbors.
-		local p2 = minetest.find_node_near(pos, 1, {"default:dirt_with_grass",
-				"default:dirt_with_dry_grass", "default:dirt_with_snow"})
+		local p2 = minetest.find_node_near(pos, 1, "group:spreading_dirt_type")
 		if p2 then
 			-- But the node needs to be under air in this case.
 			local n2 = minetest.get_node(above)
```

Comment:
I know it's not really the purpose of this PR, but `minetest.get_node(above)` is done twice in the case p2 exists but it's not air. That second call after this if block wouldn't really be needed if you get n2 already before you even check for p2.
That would be more optimal.